connectors/concur/src/test/java/com/vmware/connectors/concur/ConcurControllerTest.java

```diff
@@ -152,6 +153,22 @@ void testRequestCardsSuccess(String lang, String resFile) throws Exception {
         testRequestCards("request.json", resFile, lang);
     }
 
+    @Test
+    void testRequestCardsWithInvalidReportId() throws Exception {
+        expect(REPORT_ID_1).andRespond(withSuccess(
+                fromFile("/concur/responses/report_id_1.json")
+                        .replace("${concur_host}", mockBackend.url("")), APPLICATION_JSON));
+
+        expect(INVALID_REPORT_ID).andRespond(withStatus(HttpStatus.BAD_REQUEST));
+
+        mockConcurServer.expect(ExpectedCount.manyTimes(), requestTo("/oauth2/v0/token"))
```

Comment:
Is this actually called multiple times?  I thought the token call is only once and the same token is used for multiple report calls.

Reply:
The token call is made for every report. We can optimize by caching the token. I have created a jira ticket for the same.

https://jira-euc.eng.vmware.com/jira/browse/APF-1440

Reply:
You don't even need to cache.  Just make the first call retrieve the oauth token and pass that along and do the other calls per cards token.